hbase-server/src/main/java/org/apache/hadoop/hbase/io/hfile/bucket/BucketAllocator.java

```diff
@@ -137,10 +137,14 @@ public void addAllocation(long offset) throws BucketAllocatorException {
         if (matchFound) freeList[i - 1] = freeList[i];
         else if (freeList[i] == idx) matchFound = true;
       }
-      if (!matchFound) throw new BucketAllocatorException(
-        "Couldn't find match for index " + idx + " in free list");
+      if (!matchFound) {
+        LOG.warn("We found more entries for bucket than the bucket capacity. "
+          + "Skipping entry at cache offset {}", offset);
```

Comment:
nit: let's add more info about the bucket. Something like:

LOG.warn("We found more entries for bucket starting at offset {} for blocks of {} size. "
          + "Skipping entry at cache offset {}", baseOffset, itemAllocationSize, offset);

Reply:
ack!